### PR TITLE
Add KeyVaultFactory

### PR DESCRIPTION
**Pull request by kpajdzik**

### CODE REVIEW

File: lib/credentials/keyVaultFactory.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { DeviceTokenCredentials } from "./deviceTokenCredentials";
+import { MSIAppServiceTokenCredentials } from "./msiAppServiceTokenCredentials";
+import { MSITokenCredentials } from "./msiTokenCredentials";
+import { MSIVmTokenCredentials } from "./msiVmTokenCredentials";
+import { TokenCredentialsBase } from "./tokenCredentialsBase";
+import { UserTokenCredentials } from "./userTokenCredentials";
+import { AuthenticationContext, TokenResponse, ErrorResponse } from "adal-node";
+import { Authenticator } from "ms-rest-js";
+
+export class KeyVaultFactory {
```
Comment (daschult) [inline lib/credentials/keyVaultFactory.ts]: TypeScript generally doesn't use classes when they only contain static functions. I'd recommend removing this class and just exporting the public static functions from this file.

File: lib/credentials/keyVaultFactory.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+function _convert(credentials: MSITokenCredentials): MSITokenCredentials {
+  if (credentials instanceof MSIAppServiceTokenCredentials) {
+    return new MSIAppServiceTokenCredentials({
+      msiEndpoint: credentials.msiEndpoint,
+      msiSecret: credentials.msiSecret,
+      msiApiVersion: credentials.msiApiVersion,
+      resource: credentials.resource
+    });
+  } else if (credentials instanceof MSIVmTokenCredentials) {
+    return new MSIVmTokenCredentials({
```
Comment (RikkiGibson) [inline lib/credentials/keyVaultFactory.ts]: What does this do? Has this code just pretty much always been here?

Comment (kpajdzik) [reply lib/credentials/keyVaultFactory.ts]: Yes, I tried to optimize it but I'm a little scared without unit tests.

Comment (RikkiGibson) [reply lib/credentials/keyVaultFactory.ts]: Yeah, it's hard to test authentication code with anything besides ad-hoc testing.

**Pull request merged**